### PR TITLE
Project Scaffold and Samples

### PR DESCRIPTION
**Pull request by michaelconnor00**
Basic directory structure and samples. Also added a start of a stack for the VPC. It is missing Fargate pieces, but is ok to merge for now.

### CODE REVIEW

File: infrastructure/vpc.py [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+            enable_dns_hostnames=True,
+            enable_dns_support=True,
+            flow_logs={
+                "default":
+                    ec2.FlowLogOptions(
+                        destination=ec2.FlowLogDestination.to_cloud_watch_logs()
+                    )
+            },
+            # max_azs=99,  # Means use all AZs
+            max_azs=2,
```
Comment (cuttlefish) [inline infrastructure/vpc.py]: If we have to host our own MongoDB instance, we should plan on having to eventually make it a replica set (read: replicated cluster). I think it's recommended to have at least three instances in such a scenario. So we should probably assume we'll need at least three availability zones.

File: infrastructure/vpc.py [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+            cidr=cidr_block,
+            # configuration will create a subnet for each config, in each AZ.
+            # So us-east-1 6 public, and 6 private
```
Comment (cuttlefish) [inline infrastructure/vpc.py]: I think this comment is assume all AZs have been selected

File: README.md [hunk 3]
```diff
@@ -1,2 +1,19 @@
...
+
+All AWS deployment will be managed using AWS CDK. All stacks for application deployments are located in the `deployment/` directory, and stacks for resources or services are in `infrastructure/`. Reference the deployment [README.md](deployment/README.md) for more details.
+
+## Lambda Functions
+
+All lambda functions will be located under the `functions/` directory. Each sub directory is a lambda function. See the `functions/sample_ingest_function/` directory for a sample.
+
+## Jobs
+
+All Jobs will be located under the `jobs/` directory. Each jub directory is a job that will run in a container. See the `jobs/sample_ingest_job/` directory for a sample.
```
Comment (cuttlefish) [inline README.md:15]: Typo

**Review by cuttlefish — COMMENTED** [review]
How are we intending to handle staging/prod environments?
I'm pretty sure the intended method is to have separate AWS accounts for each environment. That doesn't seem viable for many of our use-cases.
I used a _very_ janky method on another project. @michaelconnor00, I think you haver perms for that repo: https://github.com/sparkgeo/sod-infrastructure/blob/6ced4627ade14a977fb6df5d10bf1ae7b4928f38/app.py
I used env vars to swap between configs and used different stack names to help mitigate accidentally stomping on prod. I ended up putting them in separate VPCs, but used different CIDRs so that linking between them wouldn't be too difficult if it was necessary.
It definitely isn't a very good method, but it's what I came up with at the time.

File: infrastructure/vpc.py (hunk 1 above)
Comment (michaelconnor00) [reply infrastructure/vpc.py]: Yeah, let's bump that to 3. :+1:

File: infrastructure/vpc.py (hunk 2 above)
Comment (michaelconnor00) [reply infrastructure/vpc.py]: Yes. I copied this from another project.

**Comment by michaelconnor00** [comment]
> How are we intending to handle staging/prod environments?
> I'm pretty sure the intended method is to have separate AWS accounts for each environment. That doesn't seem viable for many of our use-cases.
> I used a _very_ janky method on another project. @michaelconnor00, I think you haver perms for that repo: https://github.com/sparkgeo/sod-infrastructure/blob/6ced4627ade14a977fb6df5d10bf1ae7b4928f38/app.py
> I used env vars to swap between configs and used different stack names to help mitigate accidentally stomping on prod. I ended up putting them in separate VPCs, but used different CIDRs so that linking between them wouldn't be too difficult if it was necessary.
> It definitely isn't a very good method, but it's what I came up with at the time.

That is pretty similar to what I do. Except I explicitly create the different stack objects. For example:
```
MyStack(app, 'my-stack-prod', env=...)
MyStack(app, 'my-stack-dev', env=...)
```
Then you can run the specific stack(s) `cdk deploy my-stack-prod`. I then wrap what I need with a makefile. This is how I have seen it in the docs.

**Pull request merged**